tt_metal/impl/dispatch/device_command.hpp

```diff
@@ -15,6 +15,11 @@ class DeviceCommand {
 
     // Constants
     static constexpr uint32_t HUGE_PAGE_SIZE = 1024 * 1024 * 1024;
+
+    // COMMAND_ISSUE_REGION_SIZE and COMMAND_COMPLETION_REGION_SIZE should be configurable with some default split
+    static constexpr uint32_t COMMAND_ISSUE_REGION_SIZE = 805306368;
```

Comment:
if we want it to be configurable should we allow completion size to go to 0? and in that case EnqueueReadBuffer populates the issue region. Or should there be some min size of completion region.

Reply:
I prefer one data path (to device via issue and from device via completion) but a reduced issue region could lead to more stalls

Reply:
I think this shouldn't be a constexpr value. Rather, I think there should be some defaults when constructing the command queue, and someone those defaults propagate to send kernel to device where we pass it as a compile time arg to the producer kernel.

Reply:
how was this number calculated?

Reply:
atm just arbitrarily assigned 75% of the CQ to issue region because I figured we would be doing less read backs from device

Reply:
can we change it to `0.75 * HUGE_PAGE_SIZE` and explain psuedo-reason behind 75%

Reply:
before merging this I am planning on removing the constexpr vars representing completion/issue sizes. Instead these will be optionally user configurable. The split can be passed into `GetCommandQueue` (from https://github.com/tenstorrent-metal/tt-metal/pull/4077) 

and I will add a comment to explain the default split